service/kas/access/accessPdp.go

```diff
@@ -23,12 +23,13 @@ var decryptAction = &policy.Action{
 }
 
 type PDPAccessResult struct {
-	Access bool
-	Error  error
-	Policy *Policy
+	Access              bool
+	Error               error
+	Policy              *Policy
+	RequiredObligations []string
 }
 
-func (p *Provider) canAccess(ctx context.Context, token *entity.Token, policies []*Policy) ([]PDPAccessResult, error) {
+func (p *Provider) canAccess(ctx context.Context, token *entity.Token, policies []*Policy, obligations []string) ([]PDPAccessResult, error) {
```

Comment:
can we please name this argument like `fulfillableObligationFQNs` or something to indicate the origin (not policy, but request)?